src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/TypeManager.cs

```diff
@@ -1112,12 +1112,11 @@ internal bool GetBestAccessibleType(CSemanticChecker semanticChecker, BindingCon
                 return true;
             }
 
-            Debug.Assert(typeSrc.IsAggregateType());
+            Debug.Assert(typeSrc is AggregateType);
 
```

Comment:
It's from the initial port. There's quite a few of these where either the assertion is incorrect or the branching is pointless, perhaps in the name of defensiveness (though some would also just make an exception happen further away from the issue). I haven't traced through all the possibilities for this, so I'm not going to do anything yet except for this sort of mechanical change that clearly can't have made anything worse. Hopefully refactoring like this, along with cutting out some provably dead code as quite a few of my other open PRs do, will make that easier.